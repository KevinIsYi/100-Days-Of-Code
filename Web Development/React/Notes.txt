{Go to folder}
npx create-react-app `name`
cd `name`
npm start

rafcp 
import React from 'react'
import PropTypes from 'prop-types'

const firstApp = props => {
    return (
        <div>
            
        </div>
    )
}

firstApp.propTypes = {

}

export default firstApp