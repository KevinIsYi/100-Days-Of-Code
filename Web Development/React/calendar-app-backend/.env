PORT=4000
DB_CONNECTION=mongodb+srv://mern_user:NekxHA9lixWzsT0c@cluster0.9xvmk.mongodb.net/mern_calendar
JWT_SEED=p@L@BR@_S3CR3T@_JWT