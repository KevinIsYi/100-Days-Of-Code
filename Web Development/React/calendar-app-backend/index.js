const express = require('express');
const dotenv = require('dotenv');
const cors = require('cors');
const { dbConnection } = require('./database/config');

dotenv.config();

// Create express server
const app = express();

// Database
dbConnection();


//CORS
app.use(cors());

app.use(express.static('public'));

// Lecture and parse of body
app.use( express.json() );

app.use('/api/auth', require('./routes/auth'));
app.use('/api/events', require('./routes/events'));


// Listen petitions
app.listen(process.env.PORT, () => { // Port, callback
    console.log(`Server is running on: ${ process.env.PORT }`);
});