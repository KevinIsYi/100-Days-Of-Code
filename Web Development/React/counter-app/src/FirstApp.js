//import React, {Fragment} from 'react';
import React from 'react';
import PropTypes from 'prop-types';

//Functional components
//const FirstApp = ({greeting, subtittle}) => {
const FirstApp = ({greeting, subtittle}) => {


    return (
        <>
            <h1>{greeting}</h1>
            <p>{subtittle}</p>
        </>
    );

    /*
    return (
        // <div>
        //     <h1>Hello World!</h1>
        //     <p>Parragraph</p>
        // </div>
        <>
            <h1>Hello World!</h1>
            <p>Parragraph</p>
        </>
        // <Fragment>
        //     <h1>Hello World!</h1>
        //     <p>Parragraph</p>
        // </Fragment>
    );
    */
}

/*
FirstApp.propTypes = {
    greeting: PropTypes.string
}
*/

FirstApp.propTypes = {
    greeting: PropTypes.number.isRequired
}
FirstApp.defaultProps = {
    subtittle: "SUBTITTLE"
}


export default FirstApp;