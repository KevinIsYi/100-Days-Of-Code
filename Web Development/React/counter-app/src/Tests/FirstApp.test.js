//import { render } from "@testing-library/react";
import FirstApp from "../FirstApp.js";
import React from 'react';
import "@testing-library/jest-dom";
import { shallow } from "enzyme";

describe("Test on <FirstApp />", () => {
    // test("Must be 'Hola, soy Kevin", () => {
    //     const greeting = "Hola, soy Kevin";
    //     const {getByText} = render(<FirstApp greeting={greeting} />);

    //     expect(getByText(greeting)).toBeInTheDocument();
    // });

    test("must show <FirstApp /> correctly", () => {
        const greeting = "Hola, soy Kevin";
        const wrapper = shallow(<FirstApp greeting={greeting}/>);

        expect(wrapper).toMatchSnapshot();
    });

    test("Must show subtittle", () => {
        const greeting = "Hola, soy Kevin";
        const wrapper = shallow(
            <FirstApp 
                greeting={greeting}
                subtittle="Subtittle"
            />
        );
        const textP = wrapper.find("p").text();
        expect(textP).toBe("SUBTITTLE");
    });
});