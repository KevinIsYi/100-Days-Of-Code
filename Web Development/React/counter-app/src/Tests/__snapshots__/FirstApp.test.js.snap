// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <FirstApp /> must show <FirstApp /> correctly 1`] = `
<Fragment>
  <h1>
    Hola, soy Kevin
  </h1>
  <p>
    SUBTITTLE
  </p>
</Fragment>
`;
